flopy/mf6/utils/flopy_plugins/plugin_interface.py

```diff
@@ -0,0 +1,911 @@
+    model : MFModel
+        Model object that this plugin is a part of.
+    package : MFPackage
+        Package data that the flopy plugin for MODFLOW 6 uses to determine
+        the plugin's user settings.
+    mg : Grid
+        Model grid object for the model this plugin is a part of.
+
+    """
+
+    flopy_plugins = {}
+    interface_type = "standard"
+    abbr = None
+
+    def __init__(self):
+        self.simulation = None
+        self.model = None
+        self.package = None
+        self.mg = None
+        self._user_kwargs = None
+        self._dis_type = None
+        self.fd_debug = None
+
+    def __init_subclass__(cls):
+        """Register plugin type"""
+        super().__init_subclass__()
+        if cls.abbr is not None:
+            if cls.interface_type == "standard":
+                FPPluginInterface.flopy_plugins[cls.abbr] = cls
+
+    def receive_vars(self, simulation, model, package, user_kwargs):
+        """This method is called by MFSimulation.run_simulation prior to
+        starting the simulation.  receive_vars is called prior to init_plugin.
+        receive_vars is called to give the flopy plugin object access to the
+        simulation, model, and package objects.
+
+        Parameters
+        ----------
+        simulation : MFSimulation
+            Simulation object that this plugin is a part of. Simulation object
+            acquired when simulation is run by flopy and the receive_vars
+            method is called.
+        model : MFModel
+            Model object that this plugin is a part of.
+        package : MFPackage
+            Package data that the flopy plugin for MODFLOW 6 uses to determine
+            the plugin's user settings.
+        user_kwargs : dict
+            User defined variables passed as kwargs to run_simulation.
+        """
+        self.simulation = simulation
+        self.model = model
+        self.package = package
+        self.mg = model.modelgrid
+        self._user_kwargs = user_kwargs
+
+    def init_plugin(self, fd_debug):
+        """This method is called by MFSimulation.run_simulation prior to
+        starting the simulation and immediately after receive_vars.  Override
+        this method to execute any initialization code in your flopy plugin.
+
+        Parameters
+        ----------
+        fd_debug : file descriptor or None
+            Debug file descriptor
+
+        Returns
+        ----------
+        bool : Success/Failure
+        """
+        if fd_debug is not None:
+            self.fd_debug = fd_debug
+        return True
+
+    def sim_complete(self):
+        """This method is called by MFSimulation.run_simulation after the
+        simulation has completed.  Override this method to execute any
+        post-model processing or cleanup code in your flopy plugin.
+
+        Returns
+        ----------
+        bool : Success/Failure
+        """
+        return True
+
+    def val_from_cellid(self, cell_id, grid_array):
```

Comment:
Looks like this returns a single value. If true couldn't this function be `value_from_cellid()`.

`grid_array` can just be `array` or `arr`

Why would it ever be another `modelgrid.grid_type`? If you want to provide protection it should issue an error if it another `grid_type`.